.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+
+To suggest a new feature, talk to the IOTA community and IOTA Foundation members in the #bee-discussion channel on [Discord](https://discord.iota.org/).
+
+Or, you can submit an official [Request for Comments (RFC)](https://github.com/iotaledger/bee-rfcs/).
+
+</details>
+
+<br>
+
+<details>
+<summary>Build a new feature :hammer:</summary>
+<br>
+
+This section guides you through building a new feature. Following these guidelines helps give your feature the best chance of being approved and merged.
+
+### Before building a new feature
+
+Make sure to discuss the feature in the #bee-discussion channel on [Discord](https://discord.iota.org/).
+
+Otherwise, your feature may not be approved at all.
+
+### Building a new feature
+
+To build a new feature, check out a new branch based on the `master` branch, and be sure to consider the following:
+
+- If the feature has a public facing API, make sure to document it, using Godoc code comments
+</details>
+
+<br>
+
+<details>
+<summary>Join the Bee Initiative :deciduous_tree:</summary>
+<br>
+
+The [Bee Initiative](https://github.com/iota-community/bee) is a collaborative effort to improve the Bee developer experience by focussing on the following goals:
+
+- Quality assurance and review
+- Documentation
+- Code samples
+- Improvements to modules and libraries
+
+## How much time is involved
+
+You can invest as much or as little time as you want into the initiative.
+
+## What's in it for you
+
+In return for your time, not only do you get to be a part of the future of IOTA technology, you will also be given a badge on Discord to show others that you're a valuable member of the IOTA community.
+
+## How to join
+
+If you're interested in joining, chat to us in the #experience channel on [Discord](https://discord.iota.org/).
+
+</details>
+
+<br>
+
+<details>
+<summary>Pull requests :mega:</summary>
+<br>
+
+This section guides you through submitting a pull request (PR). Following these guidelines helps give your PR the best chance of being approved and merged.
+
+### Before submitting a pull request
+
+Before submitting a pull request, please follow these steps to have your contribution considered by the maintainers:
+
+- A pull request should have exactly one concern (for example one feature or one bug). If a PR addresses more than one concern, it should be split into two or more PRs.
+
+- A pull request can be merged only if it references an open issue
+
+    **Note:** You don't need to open an issue for minor changes such as typos, but you can if you want.
+
+- All public interfaces should have descriptive documentation, including an
+example that compiles and passes [documentation tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html)
+
+- All instances of `unsafe` should have a comment that explains why its use was unavoidable
+
+- Try to keep code comments to a minimum. If code needs to be commented to
+explain its function it can probably be refactored and be made more simple.
```

Comment:
This can get confusing. I'm not  sure if we're talking about in-doc comments or proper documentation. Every public API **should** absolutely be documented.